app/models/subscription.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Subscription < ApplicationRecord
+  validates :duration, presence: true, numericality: { only_integer: true, greater_than: 0 }
+  validate :cannot_change_after_cancelled, :cannot_change_duration, on: :update
+
+  private
```

Comment:
Should we handle these validations in setters? Or let it be handle during rails validation

Reply:
Let's keep rails validation for now!